hibernate-reactive-core/src/main/java/org/hibernate/reactive/impl/ReactiveSessionInternal.java

```diff
@@ -23,6 +24,7 @@
  *
  *  @see Stage.Session the actual user visible API
  */
+@Incubating
 public interface ReactiveSessionInternal extends org.hibernate.Session {
```

Comment:
Just a question, do we need to add the annotation here considering that this is under the package `impl`?

Reply:
Well the thing is it's exposed as a parameter in `ReactiveIdentifierGenerator`. And since it's kinda the central access point that ties everything else together, I think it eventually needs to move to a public package. But I'm not yes sure what to call that package.

Reply:
`org.hibernate.reactive.session` and `org.hibernate.reactive.session.imp`, perhaps?

Reply:
Actually I think that's a good idea.

Reply:
`org.hibernate.reactive.session` sounds good to me. What about `org.hibernate.reactive`?

Reply:
Well the problem with putting it in the root package is that it would then be the most visible API to users: apparently more important than `Stage` or `Mutiny`, which isn't correct.